keras/models.py

```diff
@@ -137,7 +137,10 @@ def get_json_type(obj):
                 weight_values = K.batch_get_value(symbolic_weights)
                 weight_names = []
                 for i, (w, val) in enumerate(zip(symbolic_weights, weight_values)):
-                    if hasattr(w, 'name') and w.name:
+                    # Default values of symbolic_weights is /variable for theano
+                    if K.backend() == 'theano' and hasattr(w, 'name') and w.name != "/variable":
+                        name = str(w.name)
+                    elif K.backend() != 'theano' and hasattr(w, 'name'):
```

Comment:
Can you make it simply:
```
if K.backend() == 'theano':
   (theano-specific code)
else:
   (non-theano specific code)
```
This is what I meant by "making it easier to keep track of parts of the code that are theano-specific".